tests/test-api.php

```diff
@@ -261,6 +262,112 @@ public function test_daily_frequency() {
 		);
 	}
 
+	/**
+	 * Custom frequency options by pageview and period.
+	 */
+	public function test_custom_frequency() {
+		$test_popup = self::create_test_popup(
+			[
+				'frequency'         => 'custom',
+				'frequency_max'     => 2,
+				'frequency_start'   => 1,
+				'frequency_between' => 2,
+				'frequency_reset'   => 'week',
+			]
+		);
+
+		// Report a pageview.
+		self::$maybe_show_campaign->save_reader_events(
+			self::$client_id,
+			[ self::create_event( [ 'post_id' => 1 ] ) ]
+		);
+
+		self::assertFalse(
+			self::$maybe_show_campaign->should_popup_be_shown( self::$client_id, $test_popup['payload'], self::$settings ),
+			'Assert not initially visible, because a frequency_start of 1 means it should only be shown on the second pageview.'
+		);
+
+		// Report another pageview.
+		self::$maybe_show_campaign->save_reader_events(
+			self::$client_id,
+			[ self::create_event( [ 'post_id' => 2 ] ) ]
+		);
+
+		self::assertTrue(
+			self::$maybe_show_campaign->should_popup_be_shown( self::$client_id, $test_popup['payload'], self::$settings ),
+			'Assert visible on the second pageview.'
+		);
+
+		// Report a view.
+		self::$report_campaign_data->report_campaign(
+			[
+				'cid'      => self::$client_id,
+				'popup_id' => Newspack_Popups_Model::canonize_popup_id( $test_popup['id'] ),
+			]
+		);
+
```

Comment:
There should be an assertion here to verify the prompt is not visible in-between two consecutive page views. It currently will fail, and so does a manual test:

1. Set up a prompt with the 'Every 4th page view…' preset
2. View four articles, observe prompt displayed
3. Refresh page – prompt is displayed on every following page load (till reaching the max of 5 views)

Reply:
This is not what I'm seeing on my end, and there is also already an assertion that tests that the prompt is not visible on two consecutive pageviews if it has a "between" value of more than 0: https://github.com/Automattic/newspack-popups/pull/908/files#diff-0eb8371b72c3645321e39c8bae29c8c4ef73c133564e8d4e09cdd952aa6f5c00R315-R318

8d9819c adds another assertion for good measure and clarifies in the code comments the difference between the pageview event and the prompt view event.

If you try this again in a new session, do you see any errors happening in the `debug` object in the API response?

Reply:
I see what you mean now! If you refresh _the same_ post, the prompt is displayed again, counting toward the max number of displays. Fixing...

Reply:
0257027 adds a check and a test to ensure that overlay prompts are NOT shown on pages that have already been viewed in the past hour. Inline prompts are still shown on consecutive views of the same page because they take up layout space.

I'm not sure if this is the right way to handle it—does it make sense to you?

Reply:
I'm not sure about the is-repeat-view approach. I can still reproduce the issue:

1. Create a prompt to be be displayed on every second pageview ("Pageviews between displays" of 1)
2. Go to homepage, then an article, then homepage – the prompt will not be visible, while it should be
3. Go to another article, then homepage again – the prompt will be visible on both consecutive page loads, which should not happen

I think the underlying issue is that the view count is only counting unique page views, which makes the `$total_views` tally inaccurate.

Reply:
@adekbadek [56f1ef4](https://github.com/Automattic/newspack-popups/pull/908/commits/56f1ef47bf867fa32e229bdf690f8451ff4c3e61) refactors a few things to handle repeat visits a bit more intentionally.

I don't think there's a perfect solution here due to the fact that we don't log repeat visits to the same pages within the same session (~1 hour). But I think this makes the most sense and also results in fewer "extra" overlay prompt displays:

* On repeat visits within the same hour, overlay prompts are not shown nor logged. So given our overlay prompt ("Pageviews between displays" of 1) if you visit homepage, then article 1, then homepage again, then article 2, then homepage again, it will only display on the first homepage visit and the article 2 visit.
* Inline prompts are displayed on repeat pageviews, but `prompt_seen` events aren't logged. So if our every-other-pageview prompt is an inline, it will be shown if you refresh or revisit a page within the hour, but it won't count toward max displays.

This does create different behavior between overlays vs inline prompts, but I think it's necessary to avoid showing overlays too often if the reader navigates through the same pages several times in a session (such as if they visit the homepage or a specific category repeatedly). The other option would be to retain all `view` events for a full month so we can go by that count instead of the reader's total view count, but that seems like a bad tradeoff because it could bloat the DB quite a lot on high-traffic sites.

Reply:
This makes sense, thanks for the detailed explanation!